homeassistant/components/lyric/climate.py

```diff
@@ -324,14 +324,23 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
                     "Could not find target_temp_low and/or target_temp_high in"
                     " arguments"
                 )
+
+            # If the device supports "Auto" mode, don't pass the mode when setting the
+            # temperature
+            mode = (
+                None
+                if device.changeableValues.mode == LYRIC_HVAC_MODE_HEAT_COOL
+                else device.changeableValues.heatCoolMode
+            )
+
             _LOGGER.debug("Set temperature: %s - %s", target_temp_low, target_temp_high)
             try:
                 await self._update_thermostat(
                     self.location,
                     device,
                     coolSetpoint=target_temp_high,
                     heatSetpoint=target_temp_low,
-                    mode=HVAC_MODES[device.changeableValues.heatCoolMode],
+                    mode=mode,
```

Comment:
I believe this was actually another bug, so fixed it while I'm at it: this was indexing the `HVAC_MODES` dict with the value of `heatCoolMode`, but the values in the `HVAC_MODES` dict are the Home Assistant strings for HVAC mode, not the Lyric strings. I don't think `heatCoolMode` can be anything other than `Heat` or `Cool`, so this was just getting lucky by the fact that the HA strings for those modes are identical. Really we don't want to do any conversion here because we're going from Lyric format to Lyric format.